### ISSUE TITLE
Several entity for an index

### ISSUE THREAD

**Issue by pengend** [algolia/search-bundle#201]
As the title says, I need to push different entities for the same index, it's impossible with the current configuration (and I had to override the default `IndexManager`) , I can do a pull request if this don't have a problem with Algolia logics.

**Comment by julienbourdeau** [comment]
I'd be happy to see your implementation and review your PR.
The main issue I found was how to know which entity is each object.

**Comment by aequasi** [comment]
A Discriminator annotation like doctrine would work, but that would be fairly coupled

**Comment by pengend** [comment]
@julienbourdeau 

What the "object" means ? My thought was to make the `indices` take array of entity or single entity. 

@aequasi you mean like in v1 ?

**Comment by pengend** [comment]
Or just disable `useAttributeAsKey` on `name`, and configure multiple entity for the same index in config, something like : 
```
algolia_search:
  indices:
    - name: persons
      class: App\Entity\Author

    - name: persons
      class: App\Entity\Visitor
```
It's quite easy to integrate that in `IndexManager`.

**Comment by julienbourdeau** [comment]
While developing the v3, I started with something like
```
algolia_search:
  indices:
    - name: persons
      classes: ['App\Entity\Author', 'App\Entity\Publisher']
```

If you use `rawSearch` it's fine, but if you use search, we recreate entities from the database, it's based on the ID you retrieved from Algolia: https://github.com/algolia/search-bundle/blob/master/src/IndexManager.php#L131-L139

If you want want to use different entities (what I called object earlier, I meant "entity class objects") you will need to:

1. change the `objectID` (today objectID is the primary key of the entity in the database)
2. find a way to differentiate what entity it is (if it's Author or Publisher)
3. split the response of `searchID` to be able to retrieve data easily with doctrine

If you only changed the `useAttributeAsKey`, you didn't run into issues?

**Comment by pengend** [comment]
Yea, when I tried this afternoon, with `rawSearch`, it's fine (even if the parameter `$className` is not great in this case), and I ran into errors with `search`, because it return an array of ids of different tables (as you said, this does not work with the current system with only one `$className` in parameters). 

I think an additional prefix mapper in configuration can differentiate entities inside an index, something like :
```
algolia_search:
  indices:
    - name: persons
      classes: App\Entity\Author
      index_prefix: author
```
the `objectID` of a such entity will be `author_1`, and we can also easily retrieve the entity from bdd by splitting the `objectID`. 

But this method may totally change the class `IndexManager`, do you anticipate that in v3 ? (btw when the v3 will be available ? )

**Comment by aequasi** [comment]
So, the opposite of this is not currently possible i think.

@julienbourdeau do you know how i could have different indecies based on a field in the entity?

EG

```yaml
algolia_search:
    indices:
        - name: Server
          class: App\Entity\Server
          enable_serializer_groups: true
        - name: Featured
          class: App\Entity\Server
          index_if: isFeatured
          enable_serializer_groups: true
        - name: Premium
          class: App\Entity\Server
          index_if: hasSubscription
          enable_serializer_groups: true
```

**Comment by julienbourdeau** [comment]
I would expect your snippet to do what you're looking for. What else would you need?

**Comment by aequasi** [comment]
The snippet only indexes servers to a single index, because the classtoIndexMapping only allows for a single class => index

**Comment by julienbourdeau** [comment]
Could you share how you want to use this? 

This should be doable with search parameters instead. In your server index, records would have a `isFeatured` and `hasSubscription` attribute, then you can search this way:

```php
$indexManager->search($query, Server::class, $em, $page, null, [
    'filters' => 'hasSubscription:true',
]);
```

**Comment by aequasi** [comment]
yeah, i could go that route too, was just interested in separating out the indexes for organizational purposes

**Comment by pengend** [comment]
I think the way to map a single class to an index is too restrictive. I can make a try to allow several classes per index, but I am a bit afraid about the impact of this to the current system.

**Comment by nunomaduro** [comment]
@pengend @aequasi @proArtex We just have made Pull Request that addresses this feature. Can you take a look and test it locally? https://github.com/algolia/search-bundle/pull/257

**Comment by nunomaduro** [comment]
This feature has been released on the version `v3.4.0`.
